### PR TITLE
586 benchmark run for every merge

### LINKED ISSUE #586: Write the first benchmark test

### PR DESCRIPTION
**Pull request by itaiad200**
closes #586

### CODE REVIEW

File: benchmarks/main.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	ch := make(chan *dto.MetricFamily)
+	go func() { _ = prom2json.ParseResponse(resp, ch) }()
+	metrics := []*dto.Metric{}
+
```
Comment (johnnyaug) [inline benchmarks/main.go:219]:
```suggestion
	ch := make(chan *dto.MetricFamily)
	go func() {
		_ = prom2json.FetchMetricFamilies(lakefsEndpoint+"/metrics", ch, nil)
	}()
	var metrics []*dto.Metric
```

File: benchmarks/main.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+				for _, label := range m.Label {
+					if *label.Name == "operation" && monitoredOps[*label.Value] {
+						metrics = append(metrics, m)
+					}
+				}
+			}
+		}
+	}
+
+	// TODO: report instead of print
```
Comment (johnnyaug) [inline benchmarks/main.go:237]: this stays for now?

**Review by johnnyaug — COMMENTED** [review]
Nice! learned a lot.
See comments. Also, there's a binary file there.

File: benchmarks/main.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		}
+	}
+}
+
+const (
+	tries        = 3
+	retryTimeout = 200 * time.Millisecond
+)
+
+func linearRetry(do func() error) error {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply benchmarks/main.go]: Done.

File: benchmarks/main.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+	go func() { _ = prom2json.ParseResponse(resp, ch) }()
+	metrics := []*dto.Metric{}
+
+	for {
+		a, ok := <-ch
+		if !ok {
+			break
+		}
+
+		if *a.Name == "api_request_duration_seconds" {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply benchmarks/main.go]: Removed.

File: benchmarks/main.go [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+			for _, m := range a.Metric {
+				for _, label := range m.Label {
+					if *label.Name == "operation" && monitoredOps[*label.Value] {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply benchmarks/main.go]: Removed.

File: benchmarks/main.go [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+var monitoredOps = map[string]bool{
+	"getObject":    true,
+	"uploadObject": true,
+}
+
+func scrapePrometheus() {
+	lakefsEndpoint := viper.GetString("endpoint_url")
+	resp, err := http.DefaultClient.Get(lakefsEndpoint + "/metrics")
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply benchmarks/main.go]: Removed.

File: benchmarks/main.go (hunk 1 above)
Comment (itaiad200) [reply benchmarks/main.go]: Removed.

File: benchmarks/main.go (hunk 2 above)
Comment (itaiad200) [reply benchmarks/main.go]: Nope, removed.

**Review by johnnyaug — APPROVED** [review]

**Pull request merged**